### PR TITLE
Very basic dynamic UI from the DAC config in the backend

### LINKED ISSUE #296: Make a configuration section DACs and have the UI match
We need to have a config array called DAC - that contains which dacs are enabled
for this instance and any settings.

One requirement - I think we will need the ability to have some DAC types listed multiple times. i.e we might have two upstream REMS - differing by URL and logins. So maybe we need DAC entries - with a DAC type (that matches to known code string?).

"rems", "ag-redcap", "manual"

The DAC page should then reflect the dacs that are configured.

For bonus points, need to think whether it would be possibly to split the config of users/password for a DAC from its entry..

"+DAC": {
  id: "my-dac",
 type: "rems"
}

and in a secret

"DAC[my-dac].username": "admin"
"DAC[my-dac].key": "abcdef"

(this might be worthy of a separate ticket- to discuss)

### PR DESCRIPTION
**Pull request by andrewpatto**

### CODE REVIEW

**Comment by andrewpatto** [comment]
REMS importer has been somewhat disabled - but that's ok as it needed a bit of a rewrite and its not in MVP scope anyhow

File: application/backend/config/dev-localhost.json5 [hunk 1]
```diff
@@ -18,6 +18,36 @@
...
+      url: "https://fake-rems.dev.umccr.org",
+      botUser: "not",
+      botKey: "arealkey",
+    },
+    {
+      id: "redcap-ag",
+      type: "redcap-australian-genomics-csv",
+      description: "Australian Genomics Redcap DAC",
+    },
+  ],
```
Comment (andrewpatto) [inline application/backend/config/dev-localhost.json5:50]: DACs now come in via config

File: application/backend/src/api/routes/internal-trpc/dataset-router.ts [hunk 2]
```diff
@@ -16,6 +16,11 @@ const inputSingleDatasetUri = z.object({
  * RPC for Datasets
  */
 export const datasetRouter = router({
+  getConfiguredDatasets: internalProcedure.query(async ({ input, ctx }) => {
+    const { user } = ctx;
+
+    return ctx.datasetService.getConfigured(user);
+  }),
```
Comment (andrewpatto) [inline application/backend/src/api/routes/internal-trpc/dataset-router.ts:23]: we need to have a discussion about configured datasets v non-configured.. but anyhow - at some point I am happy for *all* users to see a basic map of the URIs of the datasets that are available

File: application/backend/src/business/services/rems-service.ts [hunk 3]
```diff
@@ -22,21 +22,32 @@ export class RemsService {
...
+              "x-rems-user-id": d.botUser,
+            },
+          })
+          .then((a) => a.data);
+      }
+    }
+
+    return applications;
+  }
+
```
Comment (andrewpatto) [inline application/backend/src/business/services/rems-service.ts:44]: all of this needs to be refactored - so currently REMS import should be considered not working

File: application/backend/src/config/providers/provider-aws-secrets-manager.ts [hunk 4]
```diff
@@ -35,16 +36,6 @@ export class ProviderAwsSecretsManager extends ProviderBase {
...
-    // i.e. { "rems.bot": "bot" }
-    const flatSecretObject: any = JSON.parse(secretResult.SecretString);
-
-    // we can have some secrets values that are only for use elsewhere - but we don't want to
-    // trigger the 'unknown config key' of Convict - so we delete them here before passing them back
-    delete flatSecretObject["edgeDb.tlsKey"];
-    delete flatSecretObject["edgeDb.tlsCert"];
-
-    return ProviderBase.nestObject(flatSecretObject);
+    return json5.parse(secretResult.SecretString);
```
Comment (andrewpatto) [inline application/backend/src/config/providers/provider-aws-secrets-manager.ts:39]: legacy changes from the config refactor - these secret providers should just push the raw JSON5 through..

**Review by DoxasticFox — APPROVED** [review]

**Review by williamputraintan — APPROVED** [review]

**Review by mmalenic — APPROVED** [review]

**Pull request merged**